flaml/automl/model.py

```diff
@@ -1509,8 +1509,12 @@ def config2params(self, config: dict) -> dict:
         if max_depth == 0:
             params["grow_policy"] = params.get("grow_policy", "lossguide")
             params["tree_method"] = params.get("tree_method", "hist")
-        # params["booster"] = params.get("booster", "gbtree")
-        params["use_label_encoder"] = params.get("use_label_encoder", False)
+            # params["booster"] = params.get("booster", "gbtree")
+            # use_label_encoder is deprecated in 1.7.0
+            from xgboost import __version__ as xgboost_version

```

Comment:
This code block should not be under the `max_depth == 0` condition.

Reply:
should the import xgboost version be under the function then? or at the very start?